File: backend/src/modules/user/user.model.js
import mongoose from "mongoose";

const userSchema = new mongoose.Schema(
  {
    fullName: {
      type: String,
      trim: true,
    },

    email: {
      type: String,
      required: true,
      unique: true,
      lowercase: true,
    },

    password: {
      type: String,
      required: true,
    },

    profileImage: {
      type: String,
      default: "",
    },

    age: Number,

    gender: String,

    diabetesType: String,

    cgmDeviceSN: {
      type: String,
      default: "",
    },

    isProfileCompleted: {
      type: Boolean,
      default: false,
    },

    isOnboardingCompleted: {
      type: Boolean,
      default: false,
    },

    isCGMConnected: {
      type: Boolean,
      default: false,
    },

    isCGMWarmupCompleted: {
      type: Boolean,
      default: false,
    },

    warmupStartedAt: {
      type: Date,
      default: null,
    },
  },
  {
    timestamps: true,
  }
);

const User = mongoose.model("User", userSchema);

export default User;

File: mobile_app/lib/features/auth/data/models/user_model.dart
class UserModel {
  final String id;

  final String fullName;

  final String email;

  final String profileImage;

  final bool isProfileCompleted;

  final bool isOnboardingCompleted;

  final bool isCgmConnected;

  UserModel({
    required this.id,
    required this.fullName,
    required this.email,
    required this.profileImage,
    required this.isProfileCompleted,
    required this.isOnboardingCompleted,
    required this.isCgmConnected,
  });

  factory UserModel.fromJson(
    Map<String, dynamic> json,
  ) {
    return UserModel(
      id: json["_id"],

      fullName:
          json["fullName"] ?? "",

      email: json["email"] ?? "",

      profileImage:
          json["profileImage"] ?? "",

      isProfileCompleted:
          json["isProfileCompleted"] ??
              false,

      isOnboardingCompleted:
          json["isOnboardingCompleted"] ??
              false,

      isCgmConnected:
          json["isCgmConnected"] ??
              false,
    );
  }
}